tests/test_source_code_extraction.py

```
@@ -231,3 +231,27 @@ def get_source_segment(*args: Any) -> str:
+    code = """\
+import logfire
+
+def make_span(test_name: str) -> None:
+    with logfire.span('wc-test {test_name}', test_name=test_name):
+        pass
+
+for i in range(3):
+    make_span(f'test-{i}')
+"""
+    with pytest.warns(InspectArgumentsFailedWarning, match='No source code available') as record:
+        exec(code)
```

Ruff (0.16.1)

[error] Use of `exec` detected

(S102)